source/MapPanel.cpp

```diff
@@ -1383,10 +1399,46 @@ void MapPanel::DrawSystems()
 
 	// Draw the circles for the systems.
 	double zoom = Zoom();
+	BatchDrawList starBatch;
+
+	const float ringFade = mapIsStarry ? 1.5 - 1.25 * zoom : 1.;
```

Comment:
Separating `zoom` and `ringFade` with the BatchDrawList is very strange given that those two values are related and the BatchDrawList doesn't have anything to do with them.
```suggestion
	BatchDrawList starBatch;
	double zoom = Zoom();
	const float ringFade = mapIsStarry ? 1.5 - 1.25 * zoom : 1.;
```